rust/windows-client/src-tauri/src/debug_commands.rs

```diff
@@ -0,0 +1,175 @@
+//! CLI subcommands used to test features / dependencies before integrating
+//! them with the GUI, or to exercise features programmatically.
+
+use crate::prelude::*;
+use connlib_client_shared::{file_logger, Callbacks, Error, Session};
+use secrecy::SecretString;
+
+/// Test connlib and its callbacks.
+pub fn connlib(common_args: CommonArgs) -> Result<()> {
+    use connlib_client_shared::ResourceDescription;
+
+    #[derive(Clone)]
+    struct CallbackHandler {
+        handle: Option<file_logger::Handle>,
+    }
+
+    impl Callbacks for CallbackHandler {
+        type Error = std::convert::Infallible;
+
+        fn on_disconnect(&self, error: Option<&Error>) -> Result<(), Self::Error> {
+            tracing::error!("on_disconnect not implemented. Error: {error:?}");
+            Ok(())
+        }
+
+        fn on_error(&self, error: &Error) -> Result<(), Self::Error> {
+            tracing::error!("on_error not implemented. Error: {error}");
+            Ok(())
+        }
+
+        fn on_update_resources(
+            &self,
+            _resource_list: Vec<ResourceDescription>,
+        ) -> Result<(), Self::Error> {
+            tracing::error!("on_update_resources not implemented");
+            Ok(())
+        }
+
+        fn roll_log_file(&self) -> Option<PathBuf> {
+            self.handle
+                .as_ref()?
+                .roll_to_new_file()
+                .unwrap_or_else(|e| {
+                    tracing::debug!("Failed to roll over to new file: {e}");
+                    let _ = self.on_error(&Error::LogFileRollError(e));
+
+                    None
+                })
+        }
+    }
+
+    let (layer, handle) = file_logger::layer(std::path::Path::new("."));
+    setup_global_subscriber(layer);
+
+    let device_id = crate::device_id::get();
+
+    let mut session = Session::connect(
+        common_args.api_url,
+        SecretString::from(common_args.token),
+        device_id,
+        CallbackHandler {
+            handle: Some(handle),
+        },
+    )
+    .unwrap();
+
+    tracing::info!("new_session");
+
+    block_on_ctrl_c();
+
+    session.disconnect(None);
+    Ok(())
+}
+
+/// Test encrypted credential storage
+pub fn credentials() -> Result<()> {
+    use keyring::Entry;
```

Comment:
Nice. I read up on keyring a bit.

I was going to suggest using the `stronghold` plugin but its docs seem lacking. See https://github.com/tauri-apps/tauri/discussions/7846

I suppose either approach is fine since this won't have a UI component to it (thus benefitting from being a Tauri plugin)